zerver/views/upload.py

```diff
@@ -77,12 +77,10 @@ def json_upload_file(request, user_profile):
     # type: (HttpRequest, UserProfile) -> HttpResponse
     return upload_file_backend(request, user_profile)
 
-@zulip_login_required
 @has_request_variables
-def get_uploaded_file(request, realm_id_str, filename,
+def get_uploaded_file(request, user_profile, realm_id_str, filename,
                       redir=REQ(validator=check_bool, default=True)):
-    # type: (HttpRequest, str, str, bool) -> HttpResponse
-    user_profile = request.user
+    # type: (HttpRequest, UserProfile, str, str, bool) -> HttpResponse
```

Comment:
I think with this change you can just eliminate `get_uploaded_file` function and put `serve_file_backend` in urls.py (you'll need to move the `has_request_variables` part).  Might make the migration a new commit just after this one, though.